src/librustc_lint/builtin.rs

```diff
@@ -1331,23 +1356,27 @@ impl EarlyLintPass for EllipsisInclusiveRangePatterns {
                     Some(start) => format!("&({}..={})", expr_to_string(&start), end),
```

Comment:
Allocating here (and in `let end = ...`).

Reply:
Isn't this allocation okay because it's happening within the `decorate` closure?